tests/Control/UserDefinedFormControllerTest.php

```diff
@@ -367,4 +372,56 @@ public function testRecipientSubjectMergeFields()
         // check emails
         $this->assertEmailSent('test@example.com', 'no-reply@example.com', 'Email Subject: Basic Value');
     }
+
+    /**
+     * Note: will temporarily create an image in the websites public/assets/.protected folder
+     */
+    public function testImageThumbnailCreated()
+    {
+        // ensure framework is installed and a functioning website exists so there's somewhere
+        // for the uploaded file to go
+        if (!class_exists(Upload_Validator::class) || !file_exists(ASSETS_PATH)) {
+            $this->markTestSkipped("This test requires a functioning website");
+            return;
+        }
+
+        Config::modify()->set(Upload_Validator::class, 'use_is_uploaded_file', false);
+
+        $userform = $this->setupFormFrontend('upload-form');
+        $controller = UserDefinedFormController::create($userform);
+        $field = $this->objFromFixture(EditableFileField::class, 'file-field-1');
+
+        $path = realpath(__DIR__ . '/fixtures/testfile.jpg');
+        $data = [
+            $field->Name => [
+                'name' => 'testfile.jpg',
+                'type' => 'image/jpeg',
+                'tmp_name' => $path,
+                'error' => 0,
+                'size' => filesize($path),
+            ]
+        ];
+        $_FILES[$field->Name] = $data[$field->Name];
+
+        $controller->getRequest()->setSession(new Session([]));
+        $controller->process($data, Form::create());
+
+        /** @var File $image */
+        // Getting File instead of Image so that we still delete the physical file in case it was
+        // created with the wrong ClassName
+        // Using StartsWith in-case of existing file so was created as testfile-v2.jpg
+        $image = File::get()->filter(['Name:StartsWith' => 'testfile'])->last();
+        $this->assertNotNull($image);
+
+        $dbFile = $image->File;
+        $path = BASE_PATH . $dbFile->getURL();
+        $assetsDir = ASSETS_DIR;
+        $publicDir = PUBLIC_DIR;
+        $path = str_replace("/$assetsDir/", "/$publicDir/$assetsDir/.protected/", $path);
+        $path = str_replace('.jpg', '__FitMaxWzM1MiwyNjRd.jpg', $path);
+        $this->assertTrue(file_exists($path), "file did not exist in $path");
```

Comment:
```suggestion
        $store = Injector::inst()->get(AssetStore::class);
        $this->assertTrue($store->exist($image->getFilename(), $image->getHash(), 'FitMaxWzM1MiwyNjRd'));
```

Reply:
Updated to test for thumbnail variant via `AssetStore`